src/ui/components/Library/Team/View/TestRuns/TestRunList.tsx

```diff
@@ -1,19 +1,37 @@
-import { useContext } from "react";
+import { useContext, useState } from "react";
 
 import { LibrarySpinner } from "ui/components/Library/LibrarySpinner";
 import { TestRunListItem } from "ui/components/Library/Team/View/TestRuns/TestRunListItem";
+import { SecondaryButton } from "ui/components/shared/Button";
 
 import { TestRunsContext } from "./TestRunsContextRoot";
 
+const PAGE_SIZE = 50;
+
 export function TestRunList() {
-  const { testRuns } = useContext(TestRunsContext);
+  const [countToRender, setCountToRender] = useState(PAGE_SIZE);
+  const { loading, testRuns } = useContext(TestRunsContext);
+
+  const hasMore = countToRender < testRuns.length;
+  const loadMore = () => {
+    setCountToRender(Math.min(countToRender + PAGE_SIZE, testRuns.length));
+  };
+
+  if (loading) {
+    return <LibrarySpinner />;
+  }
 
   return (
     <div className="no-scrollbar m-4 flex flex-grow flex-col space-y-0 overflow-auto rounded-t-xl text-sm">
-      {testRuns ? (
-        testRuns.map((t, i) => <TestRunListItem key={i} testRun={t} onClick={() => ({})} />)
-      ) : (
-        <LibrarySpinner />
+      {testRuns.slice(0, countToRender).map(testRun => (
```

Comment:
⚠️ Important: Limit the amount of stuff we render so performance doesn't degrade over time